os_builders/README.md

```diff
@@ -1,131 +1,156 @@
-- `packfiles/rocky_sources.pkr.hcl` - Contains image definitions for Rocky (TODO)
-- `packfiles/build.pkr.hcl` - Contains the build steps for the VM image. This uses a two stage build described below
-
-CI/CD Files:
-
-- `packfiles/headless.pkrvars.hcl` - Contains the variables for doing a headless build
-
-
-Testing New OS Variants
---------------------------
-It's recommended you run this locally, so that you can see the VNC window and debug any issues:
-
-- Ensure the builder is configured, as above
-- Add your new build to the sources file, you need to add a base step and a provisioning step. See the Ubuntu file for an example.
-- Run your new/modified stage 1 build through the auto-install step: `cd packerfiles && packer build --only=stage1*<name>*` 
-
-For example:
-`cd packerfiles && packer build --only=stage1*ubuntu_2204 .`
-
-- Test the provisioning step on your new image:
-`cd packerfiles && packer build --only=stage2*<name>*`
-
-
-Prototyping new Ansible changes on a VM
-----------------------------------------
-It's recommended you use an existing VM for this testing, as it will be quicker than running an OS install and uploading :
-
-- Add your hosts to a testing inventory file, e.g. `cat inventory/testing.yml`:
-
-```
-all:
-    hosts:
-        host-172-16-255-255.nubes.stfc.ac.uk:
-            ansible_user: ubuntu  # or rocky
-```
-
-**Ensure you are on a VM!**
-
-The `provision_this_machine` variable acts as a guard from trashing your own machine. 
+The pipeline consists of the following steps:
 
-- Run the playbook
-```
-ansible-playbook -i inventory/testing.yml playbooks/provision_image.yml --extra-vars provision_this_machine=true
+- Packer pulls the latest generic image from the OS mirror into OpenStack
+- Packer generates a SSH key and uploads it to OpenStack
+- The VM is booted with the public key and uses the app creds user
+- Ansible is run to configure the VM using the [prep_user_image](prep_user_image.yml) playbook
+- Packer snapshots the VM and deletes the VM, SSH key and generic image
+
+## Setting up the environment
+
+1. Install pip, Ansible, Packer and OpenStack CLI
+  ```shell
+  # Ubuntu
+  sudo apt install python3-pip python3-venv -y
+
+  # Rocky
+  sudo dnf install python3-pip python3-venv -y
+
+  python3 -m venv image_builders
+  source image_builder/bin/activate
+  pip install -r requirements.txt
+
+  ansible-playbook prep_builder.yml
+  ```
+2. Create an applications credential (admin is only required to make images public)
+  ```shell
+  # Either place app creds in directory
+
+  mkdir -p ~/.config/openstack
+  mv clouds.yaml ~/.config/openstack/clouds.yaml
+
+  # Or
+
+  # Export credentials
+  export OS_AUTH_URL=https://<openstack_url>:5000/v3
+  export OS_APPLICATION_CREDENTIAL_ID=<app_cred_id>
+  export OS_APPLICATION_CREDENTIAL_SECRET=<app_cred_secret>
+  ```
+3. Git clone repository
+  ```shell
+  git clone https://github.com/stfc/cloud-image-builders.git
+  cd cloud-image-builders/os_builders
+  ```
```

Comment:
This step can be dropped, we're already in the repo as we needed to `requirements.txt` in step 1 (so it's implied)